hftbacktest/src/backtest/feemodel.rs

```diff
@@ -0,0 +1,54 @@
+use crate::types::Order;
+
+pub struct FeeArgs<'a> {
+    pub order: &'a Order,
+    pub amount: f64,
+}
+
+pub trait FeeModel {
+    // Calculates the fee amount.
+    fn amount(&self, args: FeeArgs) -> f64;
```

Comment:
Does `args` needs to be `FeeArgs` struct? Isn't that simple just to take `order: &Order, amount: f64`?

Reply:
I also considered that initially, but I was not sure what additional parameters might be necessary to compute fee models going forward. I didn't want to have to change the function signature every time we needed a new argument, so I encapsulated it in a separate struct, intended to be immutable.

Reply:
Even though we won't change the function signature, we still need to change the argument structure and modify the code at the call site. Since `State` is the only place where the function is called, changing it shouldn't be a problem. How about letting the function take the arguments directly and refactoring it if needed in the future?

Reply:
Done